src/sedcli_main.c

```diff
@@ -329,10 +331,33 @@ static int get_password(struct sed_key *pwd)
 	return 0;
 }
 
+static void print_sed_status(int status)
+{
+	const char *sed_status = NULL;
+
+	sed_status = sed_error_text(status);
+
+	if (sed_status == NULL && status != 0) {
+		if (status == -EINVAL) {
+			sedcli_printf(LOG_ERR, "Invalid parameter\n");
+		} else if (status == -ENODEV) {
+			sedcli_printf(LOG_ERR, "Couldn't determine device state\n");
+		} else if (status == -ENOMEM) {
+			sedcli_printf(LOG_ERR, "No memory\n");
+		} else {
+			sedcli_printf(LOG_ERR, "Unknown error\n");
+		}
+	} else {
+		if (strncmp(sed_status, SED_SUCCESS, strlen(SED_SUCCESS)) == 0)
```

Comment:
Instead of performing strncmp operation here I would expect to have enum containing all opal status codes defined in libsed.h. In this function you would simply perform arithmetic comparison instead of string comparison.